packages/cx/src/charts/Legend.js

```diff
@@ -86,11 +92,11 @@ export class Legend extends HtmlElement {
       return [list, super.renderChildren(context, instance)];
    }
 
-   renderShape(entry) {
+   renderShape(entry, instance) {
       const className = this.CSS.element(this.baseClass, "shape", {
          [`color-${entry.colorIndex}`]: entry.colorIndex != null && (isUndefined(entry.active) || entry.active),
       });
-      const shape = getShape(entry.shape || "square");
+      const shape = getShape(instance?.data?.shape || entry.shape || "square");
```

Comment:
It's safe to use `instance.data.shape`. The `data` object is always available in the rendering phase.